jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/BufferingResponseListener.java

```diff
@@ -36,7 +39,8 @@
 public abstract class BufferingResponseListener implements Listener
 {
     private final int maxLength;
-    private ByteBuffer buffer;
+    private final ByteBufferAccumulator buffer;
```

Comment:
This class is deprecated in 12.0.  In 12.1 it should probably be a `org.eclipse.jetty.io.RetainableByteBuffer.DynamicCapacity`

Perhaps it is best to rebase this PR to 12.1 and use that class.  12.1 will soon be the main branch.